src/scripts/Components/Grid.js

```diff
@@ -0,0 +1,198 @@
+      let next = reverse ? start - 1 : 1;
+      const increment = reverse ? -1 : 1;
+
+      while (next < this.matrix.length && next >= 0) {
+        if (!this.matrix[r][next]) {
+          next += increment;
+        } else {
+          if (!this.matrix[r][start]) {
+            this.matrix[r][start] = this.matrix[r][next];
+            this.matrix[r][start].setNewCoords(r, start);
+            this.matrix[r][next] = null;
+            next += increment;
+            this.checkedToAddTile = true;
+          } else if (this.matrix[r][start].value
+              === this.matrix[r][next].value) {
+            this.matrix[r][start].value *= 2;
+            this.checkedToAddTile = true;
+            this.currentScore += this.matrix[r][start].value;
+            this.currentMargeTile = this.matrix[r][start].value;
+            this.matrix[r][next].setNewCoords(r, start);
+            this.view.setTilePosition(this.matrix[r][start]);
+            this.view.setTilePosition(this.matrix[r][next]);
+            this.matrix[r][next].removeHtmlElement();
+            this.matrix[r][next] = null;
+            start += increment;
+            next += increment;
+          } else {
+            start += increment;
+
+            if (start === next) {
+              next += increment;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  moveTilesOnYAxis(reverse = false) {
+    for (let r = 0; r < this.matrix.length; r++) {
+      let start = reverse ? this.matrix.length - 1 : 0;
+      let next = reverse ? start - 1 : 1;
+      const increment = reverse ? -1 : 1;
+
+      while (next < this.matrix.length && next >= 0) {
+        if (!this.matrix[next][r]) {
+          next += increment;
+        } else {
+          if (!this.matrix[start][r]) {
+            this.matrix[start][r] = this.matrix[next][r];
+            this.matrix[start][r].setNewCoords(start, r);
+            this.matrix[next][r] = null;
+            next += increment;
+            this.checkedToAddTile = true;
+          } else if (this.matrix[start][r].value
+              === this.matrix[next][r].value) {
+            this.matrix[start][r].value *= 2;
+            this.checkedToAddTile = true;
+            this.currentScore += this.matrix[start][r].value;
+            this.currentMargeTile = this.matrix[start][r].value;
+            this.view.setTilePosition(this.matrix[start][r]);
+            this.view.setTilePosition(this.matrix[next][r]);
+            this.matrix[next][r].removeHtmlElement();
+            this.matrix[next][r] = null;
+            start += increment;
+            next += increment;
+          } else {
+            start += increment;
+
+            if (start === next) {
+              next += increment;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  moveLeft() {
+    this.moveTilesOnXAxis();
+
+    if (this.checkedToAddTile) {
+      this.addTile();
+    }
+    this.checkedToAddTile = false;
```

Comment:
You do these actions in all kinds of movements. So you could create a method for it